Classes/Service/ImportService.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SUDHAUS7\Xlsimport\Service;
+
+use Doctrine\DBAL\DBALException;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use SUDHAUS7\Xlsimport\Domain\Dto\ImportJob;
+use SUDHAUS7\Xlsimport\Event\ManipulateRelationsEvent;
+use TYPO3\CMS\Core\Core\Environment;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\DataHandling\DataHandler;
+use TYPO3\CMS\Core\Resource\Exception\FileDoesNotExistException;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Utility\StringUtility;
+
+/**
+ * @internal This class is for internal usage and no Public API
+ * Code can change in future versions
+ */
+final class ImportService
+{
+    protected EventDispatcherInterface $eventDispatcher;
+
+    public function __construct(
+        EventDispatcherInterface $eventDispatcher
+    ) {
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function isImportAllowed(string $table): bool
+    {
+        return array_key_exists($table, $GLOBALS['TCA']);
```

Comment:
This would also return true for

```php
$GLOBALS['TCA'][$table] = null;
```

I suggest to consider to do more comprehensive test here, at least checkinf for `is_array()` and additional checks.

Reply:
At the moment, added check for is array.